qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/data_selectors/cyclic_data_selector_view.py

```diff
@@ -127,6 +127,13 @@ def current_dataset_index(self) -> int:
         current_index = self.dataset_name_combo_box.currentIndex()
         return current_index if current_index != -1 else None
 
+    def set_current_dataset_index(self, index: int) -> None:
+        """Sets the index of the currently displayed dataset."""
+        if index >= self.number_of_datasets():
+            self.dataset_name_combo_box.setCurrentIndex(self.number_of_datasets()-1)
```

Comment:
Will this work if `number_of_datasets` is zero?

Reply:
I didnt have a problem when testing it. Worked fine when clearing it by changing instrument